src/couch_replicator_connection.erl

```diff
@@ -46,16 +52,16 @@ handle_call({acquire, URL}, From, State) ->
     {Pid, _Ref} = From,
     WorkerPid = case ibrowse_lib:parse_url(URL) of
         #url{host=Host, port=Port} ->
-            case ets:match_object(?MODULE, #connection{host=Host, port=Port, owner=undefined, _='_'}, 1) of
+            case ets:match_object(?MODULE, #conn{host=Host, port=Port, owner=undefined, _='_'}, 1) of
                 '$end_of_table' ->
                     {ok, WorkerPid0} = ibrowse:spawn_link_worker_process(URL),
                     couch_log:error("Worker created! ~p", [WorkerPid0]),
-                    ets:insert(?MODULE, #connection{host=Host, port=Port, owner=Pid, worker=WorkerPid0}),
+                    true = ets:insert_new(?MODULE, #conn{host=Host, port=Port, owner=Pid, worker=WorkerPid0}),
```

Comment:
yet more changes that aren't related to 'Periodically close idle connections' intent.